iModelCore/ECPresentation/Source/Hierarchies/NavNodeProviders.cpp

```diff
@@ -3538,10 +3558,9 @@ NavNodesProviderPtr SameLabelGroupingNodesPostProcessorDeprecated::_PostProcess(
 
     // attempt to find cached merged nodes provider - success means the whole hierarchy level is already post-processed and in cache.
     // it's more efficient to use the cached version compared to loading and merging everything again, so just return the cached provider.
-    DataSourceIdentifier mergedDatasourceIdentifier(GetHierarchyLevelIdentifier(*context).GetId(), {}, context->GetInstanceFilterPtr());
-    mergedDatasourceIdentifier.SetResultSetSizeLimit(context->GetResultSetSizeLimit());
-    DataSourceInfo mergedDatasourceInfo = context->GetNodesCache().FindDataSource(mergedDatasourceIdentifier, context->GetRulesetVariables());
-    if (mergedDatasourceInfo.GetIdentifier().IsValid())
+    HierarchyLevelIdentifier hlIdentifier = GetHierarchyLevelIdentifier(*context);
+    DataSourceInfo mergedDatasourceInfo = GetMergedDataSourceInfo(*context, hlIdentifier.GetId());
+    if (mergedDatasourceInfo.GetIdentifier().IsValid() && mergedDatasourceInfo.IsInitialized())

```

Comment:
Is it possible that `mergedDatasourceInfo.IsInitialized()` returns `true`, but `mergedDatasourceInfo.GetIdentifier().IsValid()` - not? Maybe only one check is enough?

Reply:
Yes, it's enough to check `IsInitialized()`